From 0a694bfda4a1cd86eea2a40c88a5fe846921f4fd Mon Sep 17 00:00:00 2001
From: Laurent Pinchart <laurent.pinchart+renesas@ideasonboard.com>
Date: Wed, 20 Mar 2013 13:14:55 +0100
Subject: [PATCH 06/15] drm/rcar-du: Fix crash when disabling an already
 disabled plane

The plane disable code tries to dereference a NULL pointer if the plane
is already disabled. Return immediately in that case.

Signed-off-by: Laurent Pinchart <laurent.pinchart+renesas@ideasonboard.com>
(cherry picked from commit 22f1261b8f2c96dd4e542bb77e06800598531bfe)

Signed-off-by: Simon Horman <horms+renesas@verge.net.au>
---
 drivers/gpu/drm/rcar-du/rcar_du_plane.c |    3 +++
 1 files changed, 3 insertions(+), 0 deletions(-)

diff --git a/drivers/gpu/drm/rcar-du/rcar_du_plane.c b/drivers/gpu/drm/rcar-du/rcar_du_plane.c
index 5055c9c..8cbd293 100644
--- a/drivers/gpu/drm/rcar-du/rcar_du_plane.c
+++ b/drivers/gpu/drm/rcar-du/rcar_du_plane.c
@@ -311,6 +311,9 @@ static int rcar_du_plane_disable(struct drm_plane *plane)
 	struct rcar_du_device *rcdu = plane->dev->dev_private;
 	struct rcar_du_plane *rplane = to_rcar_plane(plane);
 
+	if (!rplane->enabled)
+		return 0;
+
 	mutex_lock(&rcdu->planes.lock);
 	rplane->enabled = false;
 	rcar_du_crtc_update_planes(rplane->crtc);
-- 
1.7.5.4

